lib/matplotlib/gridspec.py

```diff
@@ -290,7 +290,7 @@ def subplots(self, *, sharex=False, sharey=False, squeeze=True,
             _api.warn_external(
                 "sharex argument to subplots() was an integer.  Did you "
                 "intend to use subplot() (without 's')?")
-        _api.check_in_list(["all", "row", "col", "none"],
+        _api.check_in_list(["all", "row", "col", "none", 0, 1, False, True],
```

Comment:
You've already normalized these to one of the first 4, so this isn't needed.  
```suggestion
        _api.check_in_list(["all", "row", "col", "none"],
```

Reply:
but `_api.check_in_list` is what formats the error message so they need to go it here I think.

Reply:
Ah ok.  OTOH I wouldn't list 0, 1 explicitly here since it's pretty unpythonic.

Reply:
FYI `1 in [True]` evaluates to True, as does `True in [1]` since `bool` inherits from `int`.